Offices/Offices.Application/Services/OfficeService.cs

```diff
@@ -1,6 +1,72 @@
-﻿namespace Offices.Application.Services
+﻿using System.Threading;
+using AutoMapper;
+using DnsClient.Internal;
+using Offices.Application.Models;
+using Offices.Application.Services.Abstractions;
+using Offices.Data.Entities;
+using Offices.Data.Exceptions;
+using Offices.Data.Repositories.Abstractions;
+
+namespace Offices.Application.Services
 {
-    public class OfficeService: IOfficeService
+    public class OfficeService : IOfficeService
     {
+        private readonly IGenericRepository<Office> _officeRepository;
```

Comment:
Inject specific repository, not the generic one

Reply:
Fixed